### PR TITLE
Added support for api context transformation for crumb types

### PR DESCRIPTION
**Pull request by mitsuhiko**
@getsentry/api: i'm not super happy we transform this thing on the way
out but we're doing it elsewhere too so maybe there is a reason for
doing it.

### CODE REVIEW

**Comment by mitsuhiko** [comment]
This just got rebased and requires #3013 to be merged first.

**Comment by mattrobenolt** [comment]
This is all messed up since it's attempting to merge into master, but it looks based off different branch.

File: src/sentry/interfaces/breadcrumbs.py [hunk 1]
```diff
@@ -43,97 +43,154 @@ def parse_new_timestamp(value):
...
+            if value is not None:
+                rv[key] = trim(value, 1024)
+        if not rv.get('url'):
+            raise InterfaceValidationError("No url provided for "
+                                           "'http_request' breadcrumb.")
+        return rv
+
+    def get_api_context(self, data, is_public=False):
+        rv = dict(data)
+        status_code = data.pop('status_code', None)
```
Comment (dcramer) [inline src/sentry/interfaces/breadcrumbs.py]: are there any concerns with mutating the input here?

**Comment by mitsuhiko** [comment]
This is dead now.

**Pull request closed without merging**